### PR TITLE
JDK-8301481: Replace NULL with nullptr in os/windows

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory os/windows. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301481](https://bugs.openjdk.org/browse/JDK-8301481): Replace NULL with nullptr in os/windows

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [27e135de](https://git.openjdk.org/jdk/pull/12317/files/27e135defd54e6795075ca14769a2cf1b84e458b)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12317/head:pull/12317` \
`$ git checkout pull/12317`

Update a local copy of the PR: \
`$ git checkout pull/12317` \
`$ git pull https://git.openjdk.org/jdk pull/12317/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12317`

View PR using the GUI difftool: \
`$ git pr show -t 12317`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12317.diff">https://git.openjdk.org/jdk/pull/12317.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os/windows/windbghelp.cpp [hunk 1]
```diff
@@ -287,13 +287,13 @@ void WindowsDbgHelp::print_state_on(outputStream* st) {
...
 
-    #define CHECK_AND_PRINT_IF_NULL(functionname) \
-    if (g_pfn_##functionname == NULL) { \
+    #define CHECK_AND_PRINT_IF_nullptr(functionname) \
+    if (g_pfn_##functionname == nullptr) { \
       st->print("%s" #functionname, ((num_missing > 0) ? ", " : "")); \
       num_missing ++; \
     }
 
-    FOR_ALL_FUNCTIONS(CHECK_AND_PRINT_IF_NULL)
```
Comment (jdksjolen) [inline src/hotspot/os/windows/windbghelp.cpp:296]: fix

**Review by jdksjolen — COMMENTED** [review]
Found some things that need fixing

**Review by coleenp — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]
Looks good! Thanks.

**Comment by jdksjolen** [comment]
Found a couple of bugs:

1. thread_id was assigned with NULL, but thread_id is unsigned_long on Windows. Maybe this should actually be `uintptr_t`?
2. An (ignored according to MS docs) argument had type `DWORD`, not pointer.

**Review by dholmes-ora — APPROVED** [review]
Updates look good. Thanks.

File: src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp [hunk 2]
```diff
@@ -139,7 +139,7 @@ class ZVirtualMemoryManagerSmallPages : public ZVirtualMemoryManagerImpl {
   virtual bool reserve(uintptr_t addr, size_t size) {
     const uintptr_t res = ZMapper::reserve(addr, size);
 
-    assert(res == addr || res == nullptr, "Should not reserve other memory than requested");
+    assert(res == addr || res == reinterpret_cast<uintptr_t>(nullptr), "Should not reserve other memory than requested");
```
Comment (dholmes-ora) [inline src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp:142]: Seems to me if the returned value is uintptr_t then this should be a check against zero not NULL/nullptr.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp (hunk 2 above)
Comment (stefank) [reply src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp:142]: Yes, please change this to `res == 0`.

File: src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp [hunk 3]
```diff
@@ -162,7 +162,7 @@ class ZVirtualMemoryManagerLargePages : public ZVirtualMemoryManagerImpl {
   virtual bool reserve(uintptr_t addr, size_t size) {
     const uintptr_t res = ZMapper::reserve_for_shared_awe(ZAWESection, addr, size);
 
-    assert(res == addr || res == NULL, "Should not reserve other memory than requested");
+    assert(res == addr || res == reinterpret_cast<uintptr_t>(nullptr), "Should not reserve other memory than requested");
```
Comment (stefank) [inline src/hotspot/os/windows/gc/z/zVirtualMemory_windows.cpp:165]: `res == 0`

**Review by stefank — CHANGES_REQUESTED** [review]

**Comment by jdksjolen** [comment]
Fixed the change to 0, awaiting GHA results.

**Review by dholmes-ora — APPROVED** [review]
Latest updates look good. Thanks

**Comment by jdksjolen** [comment]
All changes successful and two approvals, integrating this.

/integrate

**Pull request closed without merging**